src/components/form/EncounterForm.jsx

```diff
@@ -122,6 +122,9 @@ EncounterForm.propTypes = {
     PropTypes.func]),
   handleSubmit: PropTypes.func.isRequired,
   location: PropTypes.object,
+  manuallyExitSubmitMode: PropTypes.bool,   // defaults to false, if set to true, upon form submittal, submit mode will not exit automatically but becomes
+                                            // the responsibly of the consuming app to set this (usually in some middleware triggered by a form submitted action creator)
+                                            // (use case is when you want to prevent control from being returned to the user until after some external action is complete)
```

Comment:
Definitely looking for suggestions for a clearer name for this prop!

Reply:
I actually think this is a fine name